qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_base.py

```diff
@@ -465,14 +465,95 @@ def __init__(self, name: str, instrument: B1517A, **kwargs: Any):
         self.source_voltage = _FMTResponse(None, None, None, None)
         self._fudge: float = 1.5
 
-    def get_raw(self) -> Tuple[Tuple[float, ...], Tuple[float, ...]]:
+    def set_names_labels_and_units(self,
+                                   names: Optional[Sequence[str]] = None,
+                                   labels: Optional[Sequence[str]] = None,
+                                   units: Optional[Sequence[str]] = None
+                                   ) -> None:
         measurement_mode = self.instrument.get_measurement_mode()
-        if len(measurement_mode['channels']) != 2:
-            raise ValueError('Two measurement channels are needed, one for '
-                             'gate current and other for source drain '
-                             'current.')
+        channels = measurement_mode['channels']
 
-        smu = self.instrument.by_channel[measurement_mode['channels'][0]]
+        if names is None:
+            names = [f"param{n+1}" for n in range(len(channels))]
+            if labels is None:
+                labels = [f"Param{n + 1} Current" for n in range(len(channels))]
+
+        if labels is None:
+            labels = tuple(names)
+
+        if units is None:
+            units = ['A'] * len(names)
```

Comment:
Does this mean if no channel parameters are specified then the default channel is in current mode?

Reply:
yes, that's actually the only way this IV sweep works, it only measures currents. it's left specifyable in case someone wants to be more specific for some unknown reason :)